packages/mui-joy/src/Sheet/Sheet.tsx

```diff
@@ -0,0 +1,131 @@
+import { unstable_composeClasses as composeClasses } from '@mui/base';
+import { OverridableComponent } from '@mui/types';
+import { unstable_capitalize as capitalize } from '@mui/utils';
+import clsx from 'clsx';
+import PropTypes from 'prop-types';
+import * as React from 'react';
+import { useThemeProps } from '../styles';
+import styled from '../styles/styled';
+import { getSheetUtilityClass } from './sheetClasses';
+import { SheetProps, SheetTypeMap } from './SheetProps';
+
+const useUtilityClasses = (ownerState: SheetProps) => {
+  const { elevation, variant, color } = ownerState;
+
+  const slots = {
+    root: [
+      'root',
+      variant && `variant${capitalize(variant)}`,
+      color && `color${capitalize(color)}`,
+      elevation && `elevation${capitalize(elevation)}`,
+    ],
+  };
+
+  return composeClasses(slots, getSheetUtilityClass, {});
+};
+
+const SheetRoot = styled('div', {
+  name: 'MuiSheet',
+  slot: 'Root',
+  overridesResolver: (props, styles) => {
+    const { ownerState } = props;
+
+    return [
+      styles.root,
+      styles[ownerState.variant],
+      ownerState.variant === 'elevation' && styles[`elevation${ownerState.elevation}`],
+    ];
+  },
+})<{ ownerState: SheetProps }>(({ theme, ownerState }) => {
+  return [
+    {
+      // TODO: discuss the theme transition.
+      // This value is copied from mui-material Sheet.
+      transition: 'box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms',
+      boxShadow: theme.shadow[ownerState.elevation!],
+    },
+    theme.variants[ownerState.variant!]?.[ownerState.color!],
+  ];
+});
+
+const Sheet = React.forwardRef(function Sheet(inProps, ref) {
+  const props = useThemeProps<typeof inProps & SheetProps>({
+    props: inProps,
+    name: 'MuiSheet',
+  });
+
+  const {
+    className,
+    color = 'neutral',
+    component = 'div',
+    variant = 'contained',
```

Comment:
What's about this?

```suggestion
    component = 'div',
    variant = 'contained',
    color = variant !== 'text' ? 'neutral' : undefined,
```

- <Sheet /> : gives you white sheet with black text
- <Sheet color="primary" /> : gives you white sheet (depends on the text variant) with primary text color
- <Sheet variant="outlined" /> : gives you border & text with neutral color , if you want to change color you can add `color` prop

Reply:
> gives you white sheet with black text

This will result in grey background color when you don't pass any prop.
You can checkout the branch and play around.

I think, as you suggested earlier, having `text` as the default makes sense.